app/Services/ArticleService.php

```diff
@@ -37,6 +37,10 @@ public function getCategorizedArticlesByMagazine($magazineId)
         // For now, simply sorting categories alphabetically.
         ksort($categories);
 
+        foreach ($categories as $index => $category) {
+            $categories[$index] = collect($category)->sortByDesc('created_at');
+        }
```

Comment:
That will be still first approach, won't it?

I don't have problem with it just my experience says, it is not good approach. Single responsibility principle says one class/method must have single responsibility. I feel we are not following that principle. May be I'm taking patterns too seriously but tell me the benefit of this approach and I will accept that PR.

Lets discuss more on slack if you need direct chat.